### PR TITLE
From verification

### PR DESCRIPTION
**Pull request by Yurii666**
I will merge 2 branches

### CODE REVIEW

File: src/Stepic/Palindrome.java [hunk 1]
```diff
@@ -1,18 +1,23 @@
...
+        text1 = text.toLowerCase();
+        text1 = text1.replaceAll("[^a-zа-я0-9]","");
+        for(int i =0;i<text1.length()/2;i++){
+            if (text1.charAt(i) != text1.charAt(text1.length() - i - 1)) {
                 palin = false;
                 break;
             }
         }
         //return palin;
+        if(palin)
```
Comment (Yurii-Loienko) [inline src/Stepic/Palindrome.java:17]: Brackets should be added
if(palin) {
sout("");
} else {
sout("");
}

File: src/questions/PilindromDemo.java [hunk 2]
```diff
@@ -1,5 +1,5 @@
 package questions;
-
+// сделал все, что мог))
```
Comment (Yurii-Loienko) [inline src/questions/PilindromDemo.java:2]: very optimistic

File: src/Stepic/PrintTextPerRole.java [hunk 3]
```diff
@@ -11,6 +12,19 @@ public static void main(String[] args) {
                 "Аммос Федорович: Вот те на!",
                 "Артемий Филиппович: Вот не было заботы, так подай!",
                 "Лука Лукич: Господи боже! еще и с секретным предписаньем!"};
-        
+        String[] forSearching=new String[roles.length];
+        for(int i =0;i< roles.length;i++){
```
Comment (Yurii-Loienko) [inline src/Stepic/PrintTextPerRole.java:16]: Could you move it in one cycle?

File: src/Stepic/Palindrome.java [hunk 4]
```diff
@@ -1,18 +1,23 @@
 package Stepic;
-
+//эта задача с сайта степик
```
Comment (Yurii-Loienko) [inline src/Stepic/Palindrome.java:2]: Add conditions from the site. To make it clearer what needed to be done.

**Review by Yurii-Loienko — CHANGES_REQUESTED** [review]

File: src/Stepic/PrintTextPerRole.java (hunk 3 above)
Comment (Yurii666) [reply src/Stepic/PrintTextPerRole.java:16]: without using the additional variable forSearching?
or shove everything into the second cycle?

**Pull request closed without merging**